Source/Core/InputCommon/ControllerInterface/Device.cpp

```diff
@@ -26,45 +27,38 @@ namespace Core
 //
 Device::~Device()
 {
-  // delete inputs
-  for (Device::Input* input : m_inputs)
-    delete input;
-
-  // delete outputs
-  for (Device::Output* output : m_outputs)
-    delete output;
 }
 
-void Device::AddInput(Device::Input* const i)
+void Device::AddInput(std::unique_ptr<Input> input)
 {
-  m_inputs.push_back(i);
+  m_inputs.push_back(std::move(input));
 }
 
-void Device::AddOutput(Device::Output* const o)
+void Device::AddOutput(std::unique_ptr<Output> output)
 {
-  m_outputs.push_back(o);
+  m_outputs.push_back(std::move(output));
 }
 
 Device::Input* Device::FindInput(const std::string& name) const
```

Comment:
As far as I know, they are not used in a different thread. In fact, I don't think we even have code accessing inputs/outputs, let alone devices from more than one thread _currently_ (though devices are made shared_ptr because of #3890 which will manipulate devices from another thread)